### PR TITLE
[SYCL][Graph][L0] Reset L0 events for UR sync-points

### PR DESCRIPTION
**Pull request by EwanC**
The [Unified Runtime sync-points](https://oneapi-src.github.io/unified-runtime/core/EXP-COMMAND-BUFFER.html#sync-points) used to represent SYCL-Graph edges have a bug in the implementation of the Level Zero adapter. The L0 events used to implement the sync-points are not reset to the non-signaled state after an enqueue of the graph.

Fixed in the UR L0 adapter change https://github.com/oneapi-src/unified-runtime/pull/965 which resets the UR events to the non-signaled state at the end of the L0 command-list.

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt
Comment (EwanC) [inline sycl/plugins/unified_runtime/CMakeLists.txt:1]: The diagram of L0 command-lists in https://github.com/intel/llvm/blob/sycl/sycl/doc/design/CommandGraph.md#level-zero needs updated before this is taken out of draft

**Comment by EwanC** [comment]
CI fails on the PR look unrelated.

* From the CUDA job `ext_native_math.cpp` failure can also be seen in the [CUDA CI run](https://github.com/intel/llvm/actions/runs/6541910383/job/17764130717) from another PR #11545

* Matrix tests failures on the Arc A series can also be seen [Arc CI job](https://github.com/intel/llvm/actions/runs/6538993997/job/17756399861?pr=11312) for PR #11312

**Comment by jandres742** [comment]
> CI fails on the PR look unrelated.
> 
> * From the CUDA job `ext_native_math.cpp` failure can also be seen in the [CUDA CI run](https://github.com/intel/llvm/actions/runs/6541910383/job/17764130717) from another PR [[ESIMD] Implement unified memory API - part2 - block_load(acc, ...) #11545](https://github.com/intel/llvm/pull/11545)
> * Matrix tests failures on the Arc A series can also be seen [Arc CI job](https://github.com/intel/llvm/actions/runs/6538993997/job/17756399861?pr=11312) for PR [[UR][L0] Propagate errors from `USMAllocationMakeResident` #11312](https://github.com/intel/llvm/pull/11312)

thanks @EwanC , For https://github.com/intel/llvm/pull/11312, we are reverting those changes in https://github.com/oneapi-src/unified-runtime/pull/972. Please rebase once that is merged and relaunch.

File: sycl/plugins/unified_runtime/CMakeLists.txt
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:60]: Now that https://github.com/oneapi-src/unified-runtime/pull/965 is merged please pull in the latest `sycl` branch and update the `UNIFIED_RUNTIME_TAG` as follows:

```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 7c8617b3710736db642585fe310a64e19e059d2e
  # Merge: 20d4f49 9a13afd
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Thu Oct 26 10:20:03 2023 +0100
  #     Merge pull request #965 from Bensuo/ewan/L0_internal_event_fix
  #     [Command-buffer][L0] Reset sync-point events
  set(UNIFIED_RUNTIME_TAG 7c8617b3710736db642585fe310a64e19e059d2e)
```

**Review by mfrancepillois — APPROVED** [review]

File: sycl/doc/design/images/L0_UR_command-buffer.svg
Comment (dm-vodopyanov) [inline sycl/doc/design/images/L0_UR_command-buffer.svg:1]: What's the reason of changing svg to jpg? I think it's better to keep svg and remove jpg

Comment (EwanC) [reply sycl/doc/design/images/L0_UR_command-buffer.svg]: Good question, we changed the tool used to generate the image and just went with the default exported format. But it can export to svg as well, so updated the image back to svg format

**Review by smaslov-intel — APPROVED** [review]

File: sycl/doc/design/images/L0_UR_command-buffer.svg
Comment (dm-vodopyanov) [reply sycl/doc/design/images/L0_UR_command-buffer.svg]: The svg file looks invalid: for some reason it's actually a binary file.

Comment (EwanC) [reply sycl/doc/design/images/L0_UR_command-buffer.svg:1]: Fixed, GitHub looks to be rendering the diff of the SVG file okay now

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers Can we merge this to unblock merging the next unified-runtime PR in the queue.

**Pull request merged**
